### DISCUSSION TITLE
Investigating 1.7.1 vs 1.8.0 vs 1.8.1

### DISCUSSION THREAD

**Discussion by slinlee** [PathmindAI/nativerl#472] [General]
1.7.1 was pre-reward terms
1.8.0 introduced reward terms and reward balancing while maintaining compatibility with classic reward functions. Found to have worse performing policies in some cases
1.8.1 - enhances reward balancing and fixes some bugs.

1.8.1 is available on test.devpathmind.com, but right now we need to update the webapp to use the new 'use_auto_norm' flag

results are on this spreadsheet https://docs.google.com/spreadsheets/d/1LUpvfqk9yLMuoGngS9JZ86Oro3jSoC7bhqW3_XLRqTg/edit?usp=sharing

--- old info ---
1.7.1 - agv - classic reward function https://test.devpathmind.com/sharedExperiment/7049 the results look terrible. Throughput is 2. I think the webapp's format for setting up trainings isn't compatible with 1.7.1 anymore. (attn @ejunprung so we'll test locally??)
1.8.1 - agv - classic reward function https://test.devpathmind.com/experiment/7044

**Comment by slinlee** [comment]
I'll be updating this as I continue testing. @ejunprung

**Reply by ejunprung** [reply]
👍 I should be able to start testing as well soon.

**Reply by SaharEs** [reply]
I am testing Field Service, Amazon's model, and Felipe's model.

**Comment by ejunprung** [comment]
Btw, here is the Product Delivery results.

NativeRL 1.8.0 - https://app.pathmind.com/sharedExperiment/17265
NativeRL 1.7.1 - https://app.pathmind.com/sharedExperiment/17266

1.8.0 works well but in 1.7.1, the reward is cleared out completely (i.e. all 0s). @slinlee Do you happen to know where the reward terms stuff is in the webapp code?

**Reply by ejunprung** [reply]
Basically, I think there's two conflating issues.

1. In 1.7.1, the rewards are cleared out completely (all 0s). So the result should be terrible for all models. So yeah, as @slinlee mentioned, I'll have to train locally.
2. In 1.7.2 and above, the reward terms might be overriding the raw reward scores which means it is basically "deleting" the original reward signal. This could be the real reason for the discrepancy.

**Reply by slinlee** [reply]
Yeah. Using the current webapp will send an empty reward function where 1.7.1 expects it. So that doesn't work unfortunately.

**Reply by ejunprung** [reply]
Okay, I'm gonna do a local training now to confirm if (2) is actually a problem or not.

**Reply by ejunprung** [reply]
@slinlee Weird, in my local, the reward is all 0s as well. Are you sure the webapp clears it out? Or is it something else?

**Reply by slinlee** [reply]
you were testing 1.7.1? oh I saw your message in slack just now.

**Reply by ejunprung** [reply]
Yup

**Comment by slinlee** [comment]
Update: I created a spreadsheet because editing the table above sucked. https://docs.google.com/spreadsheets/d/1LUpvfqk9yLMuoGngS9JZ86Oro3jSoC7bhqW3_XLRqTg/edit?usp=sharing
